### DISCUSSION TITLE
skynet.ignoreret()可以将skynet.response()挂起的任务取消吗？

### DISCUSSION THREAD

**Discussion by ericzhanchina** [cloudwu/skynet#1733] [General]
实际使用中，某些协议使用了skynet.response的方式，geted将请求发给msgagent，然后等有了结果后再发给客户端。这个时候，如果客户端多发了几个这种请求， 我直接调用skynet.ignoreret()，可以将在gated里面挂起的任务给去掉掉吗？我测试好像不行，这样gated挂起的task会越来越多。那对于这种情况，是只能客户端保证请求没有回来，不要下发另外一个请求吗？

**Comment by cloudwu** [comment]
客户端和服务器的交互不应该使用 call/ret 模式，因为客户端是不可信的，不应该给客户端攻击服务器的机会。

服务器内部的交互，如果你不期待回应，就不应该用 call ；否则回应只应该区分有效回应和无效回应（超时、错误、等等），而不应该不回应。

**Reply by ericzhanchina** [reply]
但是skynet.response不就是要挂起，以后才会回应吗，客户端的消息都是经过gated转发给msgagent的，我在msgagent使用skynet.response就不会给gated回应了，这个应该是正常的吧。

**Reply by cloudwu** [reply]
客户端是不可信任的，如果你的服务器资源有可能被客户端耗尽（比如请求本身就是一种资源），那么你必须设计防范的机制。比如限制请求的并发数量。如果客户端违反了规则，你可以选择 response 错误回去，或者直接断开客户端。
